lib/mutant/mutator/node/send.rb

```diff
@@ -117,6 +118,22 @@ def emit_receiver_selector_mutations
             .each(&method(:emit_selector))
         end
 
+        # Emit mutation for `#dig`
+        #
+        # - Mutates `foo.dig(a, b)` to `foo.fetch(a).dig(b)`
+        # - Mutates `foo.dig(a)` to `foo.fetch(a)`
+        #
+        # @return [undefined]
+        def emit_dig_mutation
+          return unless selector.equal?(:dig) && arguments.any?
```

Comment:
Its not entirely pushing the negation down literally, since the negation can be consumed by the operator turning `arguments.any?` into `arguments.none?`. Where I do think we should use `arguments.empty?` as this one has less semantics (does not take a block, should also be a mutation).

The preference comes from that parsing an `unless` when the condition is composed is IMO very hard to read, especially when the negations can be pushed down into using not literally negated sub-conditions.

I think that for that reason `unless` should only be used in guard form, and than with a conditional that is not composed.